W2 reack lokman

lokman1984 opened the pull request.

janjakubnanista commented on week2/react-exercises/src/components/exercise1/Button.js:
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const Button = ({ handleButton }) => {
```
A small thing - it's good to name your props and variables well (and some say it's one of [the hardest things in programming](https://martinfowler.com/bliki/TwoHardThings.html)!).

Usually props that are event handlers start with `on` - `onSubmit`, `onFocus`, `onBlur` etc. In your case a better name for `handleButton` would be `onClick` - because it's an event handler for the `click` event.

janjakubnanista commented on week2/react-exercises/src/components/exercise1/Friend.js:
```diff
@@ -0,0 +1,34 @@
+
+const Friend = () => {
+	const [ friend, setFriend ] = useState();
+	const [ isLoadin, setLoading ] = useState(true);
```
A lil typo: `isLoading` is missing a `g` at the end :)

janjakubnanista commented on week2/react-exercises/src/components/exercise1/Friend.js:
```diff
@@ -0,0 +1,34 @@
+		fetch('https://www.randomuser.me/api?results=1')
+			.then((res) => res.json())
+			.then((data) => {
+                console.log(data)
```
And a forgotten console log :)

janjakubnanista commented on week2/react-exercises/src/components/exercise1/Friend.js:
```diff
@@ -0,0 +1,34 @@
+			.then((data) => {
+                console.log(data)
+				setFriend(data.results[0]);
+                setLoading(false);
```
It's nice to keep the code good looking, in the end you will be looking at it daily in your job. Are you familiar with ESLint, a tool that can make your code looks nice automatically?

janjakubnanista commented on week2/react-exercises/src/components/exercise1/Friend.js:
```diff
@@ -0,0 +1,34 @@
+             
+			})
+			.catch((err) => {
+				console.log('error', err);
```
Watch out for those `console.log`s :)

janjakubnanista commented on week2/react-exercises/src/components/exercise1/Friend.js:
```diff
@@ -0,0 +1,34 @@
+		<div className="Friend">
+			<Button handleButton={getFriend} />
+			{isLoadin && <p>New friend on demand by clicking the button!</p>}
+			{!hasError && friend && <FriendProfile friend={friend} />}
```
Great job, good looking component!!!! 👍

janjakubnanista commented on week2/react-exercises/src/components/exercise1/FriendProfile.js:
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+const FriendProfile = ({friend} ) => {
+	console.log(friend)
```
Oh nooooooooooo one got lost here tooo :)

janjakubnanista commented on week2/react-exercises/src/components/exercise2/Button.js:
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const Button = ({ handleButton }) => {
```
Just as I mentioned above, could you try finding a better name for this prop? Something that starts with `on` and tells us that it will be called when a user `click`s the `Button`?

janjakubnanista commented on week2/react-exercises/src/components/exercise2/DogGallery.js:
```diff
@@ -0,0 +1,37 @@
+	return (
+		<div>
+			<Button handelClick={getDogPhoto} text="Get a dog!" />
+			{isLoading && (
```
Although this works, it's a little bit difficult to understand (it's always good to write code that people will understand immediately, it saves your colleagues a lot of head-scratching :D).

This works because you set the `isLoading` to `true` initially even though you are not loading the dogs yet. If somebody now went ahead, fixed it, and set it to `false` initially, this message would not appear below the button.

You can try two things - either change the `isLoading &&` to `dogPhotos.length === 0` and the message will be shown until the first dog is loaded, or, change it to `dogPhotos.length === 0 && !isLoading` and the message will appear until the user clicks the button for the first time.

janjakubnanista commented on week2/weather-w2/src/project/Button.js:
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const Button = ({ handelButton }) => {
```
The same as above, pls see my prop naming comment

janjakubnanista commented on week2/weather-w2/src/project/Hackyourweather.js:
```diff
@@ -0,0 +1,58 @@
+			<h3>{message}</h3>
+			<form className="btn2">
+				<Search handelSearch={handelSearch} />
+				<Button handelButton={handelButton} />
```
Oooopsie small spelling mistake: `handleButton` / `handleSearch` instead of `handelButton` / `handelSearch` :)

janjakubnanista commented on week2/weather-w2/src/project/Hackyourweather.js:
```diff
@@ -0,0 +1,58 @@
+				setLoading(false);
+			}
+		} catch (err) {
+			console.log(err);
```
Forgotten `console.log` here

janjakubnanista commented on week2/weather-w2/src/project/Hackyourweather.js:
```diff
@@ -0,0 +1,58 @@
+			}
+		} catch (err) {
+			console.log(err);
+			setError('Something went wrong...');
```
You will also need to set the loading state to `false` here since the loading has stopped

janjakubnanista commented on week2/weather-w2/src/project/Hackyourweather.js:
```diff
@@ -0,0 +1,58 @@
+import Button from './Button';
+import Search from './Search';
+
+const APIKEY = process.env.REACT_APP_OPENWEATHERMAP_API_KEY;
```
Nice! It's always good to keep constants like these as far away from the code that renders UI as possible!

janjakubnanista commented on week2/weather-w2/src/project/Infoweather.js:
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+
+const Infoweather = ({ city }) => {
+	const kelvinToCelcius = (kelvin) => {
```
Good work with the temperature conversion! You can take the `kelvinToCelcius` out of the component - this way new `kelvinToCelcius` will be created every time this component renders but it will do the very same thing.

What you can do is you can put it above or below the component or even into a separate file with small utility functions (that's what you would do in larger code bases)

wouterkleijn commented:
Thanks @janjakubnanista for the review. Can you have a look @lokman1984 ?

lokman1984 commented:
> Thanks @janjakubnanista for the review. Can you have a look @lokman1984 ?

hi 

> Thanks @janjakubnanista for the review. Can you have a look @lokman1984 ?

hi @wouterkleijn @janjakubnanista  thank you for feedback.i corrected it and re uploaded it,but unfortunately ,a mistake occurred in week 3 with github ,i deleted the fork an reconnected .so i corrected and reupload homework week2 full exercises :)
